decisions/0004-host-naming-conventions.md

```diff
@@ -0,0 +1,57 @@
+---
+status: proposed
+date: 2022-12-11
+deciders: Shreyas Cholia
+consulted: {list everyone whose opinions are sought (typically subject-matter experts); and with whom there is a two-way communication}
+informed: {list everyone who is kept up-to-date on progress; and with whom there is a one-way communication}
+---
+# Standardize convention for host names in NMDC
+
+## Context and Problem Statement
+
+Hostnames in the `*.microbiomedata` should have a standard naming pattern in order to 
+- make it easier for humans to remember endpoints
+- enable automation by using a standard pluggable convention 
+- integrate with underlying kubernetes / rancher platform
+- simplify management of certificates
+
+## Decision Drivers
+
+* Restrictions imposed by Spin - Spin has internal names and we would like parity with these if possible
+* Provisioning of certificates and ensuring that we retain flexibility in doing so
+* Names supported by cloudflare moving forwards
+
+## Considered Options
+We would like to capture the service name (data, api, db etc.) and the environment (production, sandbox, dev) in the hostname followed by `microbiomedata.org` For the canonical production host we can drop the environment to make it simpler for public user access
+
+* `{service}-{env}` with `-` as the separator 
+    - eg. `data-sandbox.microbiomedata.org`
+* `{service}.{env}` with `.` as the separator 
+    - eg. `data.sandbox.microbiomedata.org`
+    - This introduces a potential extra subdomain in DNS depending on the hosting provider
+
+We will use the following service names for public facing services (This list can be expanded))
+* `api` - API service
+* `data` - Data Portal  
+
+We will use the following environment names (This list can be expanded)
+* `prod` - production (optional since this is the user facing instance)
+* `dev` - development 
```

Comment:
some expanded scope note here for each would be helpful. what distinguishes "dev" vs "sandbox"?

Reply:
See updates